server/certstore/certstore_windows.go

```diff
@@ -137,9 +140,38 @@ type winPSSPaddingInfo struct {
 	cbSalt   uint32
 }
 
+// createCACertsPool generates a CertPool from the Windows certificate store,
+// adding all matching certificates from the caCertsMatch array to the pool.
+// All matching certificates (vs first) are added to the pool based on a user
+// request. If no certificates are found an error is returned.
+func createCACertsPool(cs *winCertStore, storeType uint32, caCertsMatch []string) (*x509.CertPool, error) {
+	var errs []error
+	var count int
+	caPool := x509.NewCertPool()
+	for _, s := range caCertsMatch {
+		lfs, err := cs.caCertsBySubjectMatch(s, storeType)
+		if err != nil {
+			if errs == nil {
+				errs = make([]error, 0)
+			}
+			errs = append(errs, err)
+		} else {
+			for _, lf := range lfs {
+				caPool.AddCert(lf)
+				count++
+			}
+		}
+	}
+	// If no certs were added, return the errors.
+	if count == 0 {
```

Comment:
No need for a `count`, doing a `len(errs)` would suffice.

Reply:
Agree - that's cleaner - if len(errs) == len(caCertsMatch)...